### DISCUSSION TITLE
Soundtouch inside crossfade ?

### DISCUSSION THREAD

**Discussion by Russsgithub** [savonet/liquidsoap#2466] [Q&A]
I'm trying to use soundtouch or stretch inside a cross function to achieve beat syncing of a source. The code I'm using is something like this.
```
s = request.dynamic('./script/path')

def transition(a, b)
   add([fade.out(duration=5., a.source), fade.in(duration=5., soundtouch(tempo=2., b.source))])

end
   cross(transition=transition, s)
```

The crossfade works as expected, but as soon as it ends soundtouch no longer effects the source and a sudden speed change happens sending the source (b) back to it's original speed.

I've tried soundtouch on the source before the crossfade. e.g.
```
s = request.dynamic('./script/path')
s = soundtouch(tempo=2., s)

def transition(a, b)
   add([fade.out(duration=5., a.source), fade.in(duration=5., b.source)])

end
   cross(transition=transition, s)
```
In this case cross doesn't trigger at all , I think i've noticed before that soundtouch pre crossfade doesn't work as I'd like it to.

I need to not effect both a and b so soundtouch after cross wouldn't work or me.

I can't see a solution for this if cross is behaving as it's surposed to.

Any help is useful.

Thanks.

**Comment by millicent-audio-mirror** [comment]
It seems to me that stretch / soundtouch remove track markers from the source, which is stopping cross from seeing the track transitions.

If I use

```
s = source
s = soundtouch(tempo=0.5, s)
s.on_track(fun(m) -> print("new track"))
...
```
I get the new track line printed at the beginning of the stream, and then not again.

**Comment by Russsgithub** [comment]
I am using liq 2.2 (github main) with this test. FYI

**Comment by toots** [comment]
We are aware of this and plan on addressing it with the changes from the ongoing `2.2` release. You can track the issue here: https://github.com/savonet/liquidsoap/issues/2467
